include/envoy/upstream/upstream.h

```diff
@@ -123,43 +123,72 @@ class Host : virtual public HostDescription {
 
 typedef std::shared_ptr<const Host> HostConstSharedPtr;
 
+typedef std::vector<HostSharedPtr> HostVector;
+typedef std::shared_ptr<HostVector> HostVectorSharedPtr;
+typedef std::shared_ptr<const HostVector> HostVectorConstSharedPtr;
+
+/**
+ * Bucket hosts by locality.
+ */
+class HostsPerLocality {
+public:
+  virtual ~HostsPerLocality() {}
+
+  /**
+   * @return bool is local locality one of the locality buckets?
```

Comment:
nit: Might comment that if true, it will be the first element in the vector returned by `get()`